### PR TITLE
GIVCAMP-192: Add basic auth plugin

### PR DESCRIPTION
**Pull request by sherakama**
# READY FOR REVIEW

# Summary
- Adds (browser based) basic auth to all contexts

# Review By (Date)
- Whenever

# Criticality
- 🤷 

# Review Tasks

1. Navigate to the [build preview](deploy-preview-144--giving-campaign.netlify.app)
2. Do not enter anything for username
3. Enter `0nPurrpose` for password

### CODE REVIEW

**Comment by sherakama** [comment]
@yvonnetangsu You'll see both Netlify and Browser auth right now. Once approved and deployed everywhere, I will remove the Netlify auth.

**Comment by yvonnetangsu** [comment]
@sherakama Not sure if I'm looking at this right. For some reason, I'm still only seeing the original Netlify auth and not the browser one (like on the Giving site). It let me in after I entered the password in the Netlify auth 🤔 Tried again in another browser and still the same. Perhaps we can look at it together in our 1:1 😄

**Comment by sherakama** [comment]
>  I'm still only seeing the original Netlify auth and not the browser one

That might be because I am using the same creds for both. You can try by removing the Netlify auth and using an incognito window to try again.

**Review by yvonnetangsu — APPROVED** [review]
Thank you @sherakama ! I'm going to merge this and then can disable the Netlify auth 😄

**Comment by yvonnetangsu** [comment]
@sherakama I've removed the Netlify auth and I'm using an incognito browser with my new PR preview, but don't seem to see the browser auth 🤔 
https://deploy-preview-149--giving-campaign.netlify.app/

**Comment by sherakama** [comment]
Huh, lemme double check something.

File: plugins/netlify-plugin-contextual-auth/manifest.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+name: netlify-plugin-contextual-basic-auth
+inputs:
+  - name: username
+    description: Username for http basic auth
+    required: true
```
Comment (yvonnetangsu) [inline plugins/netlify-plugin-contextual-auth/manifest.yml:5]: Does username required need to be changed to `false`?

**Comment by sherakama** [comment]
@yvonnetangsu 

https://dev--giving-campaign.netlify.app/

Two straight commits:
https://github.com/SU-SWS/ood-giving-campaign/commit/0d672544d657735f9aa42336a08b9b72206e5080
https://github.com/SU-SWS/ood-giving-campaign/commit/c5c3435920bfc57dd99b09f035e6d2098ac839ad

1. Write to the correct place so that Netlify picks it up
2. Add a username because an empty name didn't work

**Comment by yvonnetangsu** [comment]
> @yvonnetangsu
> 
> https://dev--giving-campaign.netlify.app/
> 
> Two straight commits: [0d67254](https://github.com/SU-SWS/ood-giving-campaign/commit/0d672544d657735f9aa42336a08b9b72206e5080) [c5c3435](https://github.com/SU-SWS/ood-giving-campaign/commit/c5c3435920bfc57dd99b09f035e6d2098ac839ad)
> 
> 1. Write to the correct place so that Netlify picks it up
> 2. Add a username because an empty name didn't work

Thank you for investigating @sherakama ! Working as expected now 🎉 😄 I'll post a comment to let the team know that a username is now required.

**Pull request merged**